logic/peer.js

```diff
@@ -175,7 +182,9 @@ Peer.prototype.properties = [
 	'httpPort',
 ];
 
-Peer.prototype.immutable = ['ip', 'wsPort', 'httpPort', 'string'];
+Peer.prototype.required = ['ip', 'wsPort', 'string'];
+
+Peer.prototype.optional = ['httpPort', 'nonce'];
```

Comment:
Is it really required to make them a prototype property? If I understand right, this way both arrays will be associated/created for every instance of `Peer` object. 

Instead why not just a module level private member.

Reply:
@nazarhussain thats a good point, we can make them private members of the modules, since, we are accessing these properties through peer instance in tests we might need to keep them the way it is now.